### ISSUE TITLE
LLM trading - sentiment analysis

### ISSUE THREAD

**Issue by kaidatavis** [Vis4Sense/student-projects#61]

**Comment by kaidatavis** [comment]
### Sentiment benchmark

(please create a table to compare the different options following the format in #59 and #60)

**Comment by kaidatavis** [comment]
### Sentiment performance of different LLMs 

(please create a table to compare the different options following the format in #59 and #60)

**Comment by erwin27** [comment]
LLMs Benchmark on Financial News Dataset

Test dataset: Financial Phrasebank (https://huggingface.co/datasets/financial_phrasebank)
English sentences from financial news, and classified as either positive, negative, or neutral by researchers knowledgeable in the finance domain.

This benchmark using 1,000 records sample (random seed = 42) from "sentence_allagree" subset. 
The distribution of the test set is same as the subset population (Neutral: 61.4%, Positive: 25,2%, Negative: 13,4%)

LLMs Model tested are from gpt4all.io (4-bit Quantization)

Hardware (Laptop) specs used for test:
Windows 11
CPU: 12th Gen Intel i7-12700H 4.70 GHz
RAM: 32GB 4800Mhz DDR5
GPU: Nvidia RTX 3070 Ti 8Gb
Storage: 1TB M.2 SSD 


| Model | Accuracy | Run Time | avg time/iter |Spec n requirement: Param/size/RAM|
| ------------ | ---- | ---- | ---- |---------|
| gpt4all-13b-snoozy-q4_0 |0.3670|5hrs 37min|20s|13b/6.86 GB/16 GB|
| nous-hermes-llama2-13b.Q4_0  |   |   |   |13b/6.86 GB/16 GB|
| wizardlm-13b-v1.2.Q4_0 |0.8170|5hrs 36min|20s|13b/6.86 GB/16 GB|
| orca-2-13b.Q4_0 |   |   |   |13b/6.86 GB/16 GB|
| orca-2-7b.Q4_0 |   |   |   |7b/3.56 GB/8 GB|
| mistral-7b-openorca.Q4_0 |   |   |   |7b/3.83 GB/8 GB|
| mistral-7b-instruct-v0.1.Q4_0 |   |   |   |7b/3.83 GB/8 GB|
| gpt4all-falcon-q4_0 |   |   |   |7b/3.92 GB/8 GB|
| mpt-7b-chat-merges-q4_0 |   |   |   |7b/3.54 GB/8 GB|
| orca-mini-3b-gguf2-q4_0 |   |   |   |3b/1.84 GB/4 GB|

**Comment by kaidatavis** [comment]
@erwin27, this is a very good start. Please keep filling the table as more results become available. 

Did you have a chance to discuss with Xiruo about distribute the work? Maybe she can test some of the models while you are testing the others?

Finally, it will be good to test the 'strength' of the sentiment besides positive, negative, and neutral. For example, there can be 'very positive' and 'a little positive', which translate to 1.8 or 1.2 sentiment score if 2 is the most positive.

**Comment by erwin27** [comment]
@kaidatavis , yes me and Xiruo have discussed several times already since monday. Sure, we will  we will continue until the rest of the model and may be finding another financial related dataset if possible. We also will try your recommendation for the sentiment analysis strength as that would be huge impact for the project if  latter we implement it. Thank you @kaidatavis
